src/Package/Impl/DataInspect/Viewers/FunctionViewer.cs

```diff
@@ -0,0 +1,86 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See LICENSE in the project root for license information.
+
+using System;
+using System.ComponentModel.Composition;
+using System.IO;
+using System.Threading.Tasks;
+using EnvDTE;
+using Microsoft.Common.Core;
+using Microsoft.R.Core.Formatting;
+using Microsoft.R.Debugger;
+using Microsoft.R.Editor.Settings;
+using Microsoft.R.Editor.Signatures;
+using Microsoft.R.Host.Client;
+using Microsoft.VisualStudio.R.Package.Repl;
+using Microsoft.VisualStudio.R.Package.Shell;
+using static System.FormattableString;
+
+namespace Microsoft.VisualStudio.R.Package.DataInspect.Viewers {
+    [Export(typeof(IObjectDetailsViewer))]
+    internal sealed class FunctionViewer : IObjectDetailsViewer {
+        private readonly IRSessionProvider _sessionProvider;
+        private readonly IDataObjectEvaluator _evaluator;
+
+        [ImportingConstructor]
+        public FunctionViewer(IRSessionProvider sessionProvider, IDataObjectEvaluator evaluator) {
+            _sessionProvider = sessionProvider;
+            _evaluator = evaluator;
+        }
+
+        #region IObjectDetailsViewer
+        public ViewerCapabilities Capabilities => ViewerCapabilities.Function;
+
+        public bool CanView(DebugValueEvaluationResult evaluation) {
```

Comment:
Since it is a MEF export it can also be imported somewhere else. So this is basically like command GetStatus and Invoke - no matter if you return 'no supported' Invoke can still be called.